hera_pspec/tests/test_pspecdata.py

```diff
@@ -932,12 +982,41 @@ def test_get_H(self):
                 self.ds.set_taper(taper)
 
                 self.ds.set_Ndlys(Nfreq//3)
-                H = self.ds.get_H(key1, key2)
+                H = self.ds.get_H(key1, key2, average_times=True)
                 self.assertEqual(H.shape, (Nfreq//3, Nfreq//3)) # Test shape
 
                 self.ds.set_Ndlys()
-                H = self.ds.get_H(key1, key2)
-                self.assertEqual(H.shape, (Nfreq, Nfreq)) # Test shape
+                H = self.ds.get_H(key1, key2, average_times=True)
+
+    def test_get_H_fft(self):
+        """
+        Test Fisher matrix calculation with FFT.
+        """
+        self.ds = pspecdata.PSpecData(dsets=self.d, wgts=self.w)
+        Nfreq = self.ds.Nfreqs
+        multiplicative_tolerance = 1.
+        key1 = (0, 24, 38)
+        key2 = (1, 25, 38)
+
+        for input_data_weight in ['identity','iC', 'dayenu']:
+            self.ds.set_weighting(input_data_weight)
+            if input_data_weight == 'dayenu':
+                nt.assert_raises(ValueError,self.ds.R, key1)
+                rpk = {'filter_centers':[0.],'filter_half_widths':[200e-9],'filter_factors':[1e-9]}
+                self.ds.set_r_param(key1,rpk)
+                self.ds.set_r_param(key2,rpk)
+            for taper in taper_selection:
+                self.ds.set_taper(taper)
+
+                self.ds.set_Ndlys(Nfreq)
+                H1 = self.ds.get_H(key1, key2, average_times=True, sampling=True)
```

Comment:
Would be best to explicitly put `allow_fft=False` here in case default ever changes

Reply:
Also: Are there tests in here to handle the new caching setup in `set_G` etc etc?

Reply:
I've added `allow_fft=False`. I've also added some tests that the expected number of H and G matrices are being cached with each change in configuration in `test_get_H`, `test_get_G`, `test_get_M`, and `test_get_W`.